core/DataTable.php

```diff
@@ -1214,25 +1239,63 @@ public function getSerialized($maximumRowsInDataTable = null,
                 $row->removeSubtable();
             }
         }
-        // we load the current Id of the DataTable
-        $forcedId = $this->getId();
 
         // if the datatable is the parent we force the Id at 0 (this is part of the specification)
         if ($depth == 0) {
             $forcedId = 0;
+            $subtableId = 0;
         }
 
         // we then serialize the rows and store them in the serialized dataTable
-        $addToRows = array(self::ID_SUMMARY_ROW => $this->summaryRow);
+        $rows = array();
+        foreach ($this->rows as $id => $row) {
+            if (array_key_exists($id, $consecutiveSubtableIds)) {
+                $backup = $row->subtableId;
+                $row->subtableId = $consecutiveSubtableIds[$id];
+                $rows[$id] = $row->export();
+                $row->subtableId = $backup;
+            } else {
+                $rows[$id] = $row->export();
+            }
+        }
 
-        $aSerializedDataTable[$forcedId] = serialize($this->rows + $addToRows);
```

Comment:
We do no longer serialize the whole `Row` instances, instead only an array for each row which is returned by `export()`. This is not only faster but allows us to refactor code in the future as we are no longer dependent on the `Row` implementation.